app/models/line_item.rb

```diff
@@ -0,0 +1,25 @@
+class LineItem < ApplicationRecord
+  belongs_to :order
+  belongs_to :deal
+
+  before_create :calculate_loyality_discount_percentage_and_price
```

Comment:
let's do this on before_validation so validations can be applied on these fields later if needed

Reply:
Done